compiler/bindings/python/test/ir/dialects_test.py

```diff
@@ -391,3 +391,70 @@ def compilation_info():
     assert compilation_info is not None
     assert compilation_info.lowering_config == lowering_config
     assert compilation_info.translation_info == translation_info
+
+
+@run
+def gpu_target_info_attribute_parsing():
+    mlir_string = """
+    hal.executable private @main_dispatch_0 {
+        hal.executable.variant public @rocm_hsaco_fb
+            target(<"rocm", "rocm-hsaco-fb",
+                {
+                abi = "hip",
+                iree_codegen.target_info = #iree_gpu.target<
+                    arch = "gfx942",
+                    features = "",
+                    wgp = <
+                    compute = fp64|fp32|fp16,
+                    storage = b64|b32|b16|b8,
+                    subgroup = shuffle|arithmetic,
+                    dot = dp4xi8toi32,
+                    mma = [<MFMA_F32_16x16x4_F32>],
+                    subgroup_size_choices = [64],
+                    max_workgroup_sizes = [1024, 1024, 1024],
+                    max_thread_count_per_workgroup = 1024,
+                    max_workgroup_memory_bytes = 65536,
+                    max_workgroup_counts = [2147483647, 2147483647, 2147483647],
+                    max_load_instruction_bits = 128,
+                    simds_per_wgp = 4,
+                    vgpr_space_bits = 16384
+                    >
+                >
+                }>
+            ) {
+        }
+    }
+    """
+
+    module = ir.Module.parse(mlir_string)
+    variant_op_list = iree_codegen.get_executable_variant_ops(module)
+    assert len(variant_op_list) == 1, "Expect one executable variant op"
+    variant_op = variant_op_list[0]
+    executable_variant_op = variant_op.opview
+    target = executable_variant_op.target
+    gpu_target_info = iree_gpu.get_gpu_target_info(target)
+
+    arch = gpu_target_info.arch
+    assert arch == "gfx942", f"Expected arch 'gfx942', got '{arch}'"
```

Comment:
I think failed assertions are going to print the actual and expected values -- could you check if this is the case, and remove those messages if it works like I described? Also below.